controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -224,38 +230,85 @@ func (r *RegisteredClusterReconciler) syncSyncTarget(computeContext context.Cont
 
 		locationContext := logicalcluster.WithCluster(computeContext, logicalcluster.New(regCluster.Spec.Location))
 
-		syncTargetLength, err := r.checkSynctargetExists(locationContext, regCluster)
+		syncTarget, err := r.getSyncTarget(locationContext, regCluster)
 		if err != nil {
 			return giterrors.WithStack(err)
 		}
-		if syncTargetLength == 0 {
+
+		labels := map[string]string{
+			RegisteredClusterNamelabel:      regCluster.Name,
+			RegisteredClusterNamespacelabel: regCluster.Namespace,
+			RegisteredClusterWorkspace:      strings.ReplaceAll(regCluster.Annotations["clusterName"], ":", "-"),
+			RegisteredClusterUidLabel:       string(regCluster.UID),
+		}
+		for k, v := range regCluster.Labels {
+			labels[k] = v
+		}
```

Comment:
this would overlap these 
```
labels := map[string]string{
			RegisteredClusterNamelabel:      regCluster.Name,
			RegisteredClusterNamespacelabel: regCluster.Namespace,
			RegisteredClusterWorkspace:      strings.ReplaceAll(regCluster.Annotations["clusterName"], ":", "-"),
			RegisteredClusterUidLabel:       string(regCluster.UID),
		}
```